docs/source/philosophy.rst

```diff
@@ -0,0 +1,65 @@
+VISSL Philosophy
```

Comment:
wohoo! we need to index this file under https://github.com/facebookresearch/vissl/blob/master/docs/source/index.rst#vissl-documentation in `NOTES` :).

Reply:
In 801c2883c8a0bab4ee4ad6016608cd6c9b380107, I ended up removing the philosophy documentation and instead move it where it seems to be more important: together with the documentation on how to extend VISSL with new data sources (where it seems to belong)